### PR TITLE
Implement fine-grained errors to cloud queries

### PR DESCRIPTION
**Pull request by underhood**
##### Summary
Implements #12340 

##### Test Plan
See the generated message. Cloud hasn't implemented this yet but we got green light to send this (Cloud should for now ignore until they find time to implement) but should not cause errors on cloud.

### CODE REVIEW

File: aclk/aclk_util.h [hunk 1]
```diff
@@ -5,6 +5,19 @@
 #include "libnetdata/libnetdata.h"
 #include "mqtt_wss_client.h"
 
+#define CLOUD_EC_MALFORMED_NODE_ID   1
+#define CLOUD_EMSG_MALFORMED_NODE_ID "URL requests node_id but there is not enough chars following (for it to be valid uuid)."
+#define CLOUD_EC_NODE_NOT_FOUD       2
+#define CLOUD_EMSG_NODE_NOT_FOUD     "Node with requested node_id not found"
```
Comment (stelfrag) [inline aclk/aclk_util.h:11]: `FOUD` begs for a spellcheck sooner or later :smile: Also this message should probably be just "Node id not found"  or "Node does not exist"

Comment (underhood) [reply aclk/aclk_util.h]: thanks

File: aclk/aclk_query.c [hunk 2]
```diff
@@ -115,20 +115,20 @@ static int http_api_v2(struct aclk_query_thread *query_thr, aclk_query_t query)
...
-            aclk_http_msg_v2_err(query_thr->client, query->callback_topic, query->msg_id, w->response.code, NULL, 0);
+            aclk_http_msg_v2_err(query_thr->client, query->callback_topic, query->msg_id, w->response.code, CLOUD_EC_MALFORMED_NODE_ID, CLOUD_EMSG_MALFORMED_NODE_ID, NULL, 0);
             goto cleanup;
         }
         strncpyz(nodeid, node_uuid, UUID_STR_LEN - 1);
 
         query_host = node_id_2_rrdhost(nodeid);
         if (!query_host) {
-            error("Host with node_id \"%s\" not found! Returning 404 to Cloud!", node_uuid);
+            error_report("Host with node_id \"%s\" not found! Returning 404 to Cloud!", node_uuid);
```
Comment (stelfrag) [inline aclk/aclk_query.c:128]: The `node_uuid` is not actually the node id. It is the entire query starting with the node id. So the error message ends up logging something like: 
```
Host with node_id "67c22c43-b604-40fc-b60d-4dc53f83edef/api/v1/data?after=1647773828&before=1647774728&context=system.io&dimensions=out&format=json&group=average&gtime=0&options=absolute%7Callow_past%7Cflip%7Cjsonwrap%7Cms&points=40" not found! Returning 404 to Cloud!
```

**Review by MrZammler — APPROVED** [review]

**Review by stelfrag — APPROVED** [review]

File: aclk/aclk_query.c [hunk 3]
```diff
@@ -115,20 +115,20 @@ static int http_api_v2(struct aclk_query_thread *query_thr, aclk_query_t query)
         char *node_uuid = query->data.http_api_v2.query + strlen(NODE_ID_QUERY);
         char nodeid[UUID_STR_LEN];
         if (strlen(node_uuid) < (UUID_STR_LEN - 1)) {
-            error("URL requests node_id but there is not enough chars following. Returning 404 to Cloud.");
+            error_report(CLOUD_EMSG_MALFORMED_NODE_ID);
             retval = 1;
             w->response.code = 404;
-            aclk_http_msg_v2_err(query_thr->client, query->callback_topic, query->msg_id, w->response.code, NULL, 0);
+            aclk_http_msg_v2_err(query_thr->client, query->callback_topic, query->msg_id, w->response.code, CLOUD_EC_MALFORMED_NODE_ID, CLOUD_EMSG_MALFORMED_NODE_ID, NULL, 0);
```
[the remark this answers is not in the document]
Comment (underhood) [reply aclk/aclk_query.c:121]: Yeah should be easier to change this in future after cloud consumes the new custom error codes -> so they can match it whenif we change from 404

**Pull request merged**